### PR TITLE
feat(batch,agg): support `array_agg` for batch mode

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR adds support for `array_agg` in batch mode.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#4657

### CODE REVIEW

File: src/expr/src/vector_op/agg/array_agg.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        Ok(())
+    }
+
+    fn update_multi(
+        &mut self,
+        input: &DataChunk,
+        start_row_id: usize,
+        end_row_id: usize,
+    ) -> Result<()> {
+        for row_id in start_row_id..end_row_id {
```
Comment (BugenZhao) [inline src/expr/src/vector_op/agg/array_agg.rs:66]: How about reserving some capacity for `self.values` before pushing one-by-one?

File: src/expr/src/vector_op/agg/array_agg.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+            self.update_single(input, row_id)?;
+        }
+        Ok(())
+    }
+
+    fn output(&mut self, builder: &mut ArrayBuilderImpl) -> Result<()> {
```
Comment (BugenZhao) [inline src/expr/src/vector_op/agg/array_agg.rs:72]: Oops. Are we still using `RwResult` for aggregators? 🥵

**Review by BugenZhao — APPROVED** [review]
LGTM

File: src/expr/src/vector_op/agg/array_agg.rs (hunk 2 above)
Comment (BugenZhao) [reply src/expr/src/vector_op/agg/array_agg.rs:72]: Opened a PR for this: #4873

File: src/expr/src/vector_op/agg/array_agg.rs (hunk 1 above)
Comment (richardchien) [reply src/expr/src/vector_op/agg/array_agg.rs:67]: ok

**Review by liurenjie1024 — APPROVED** [review]

**Pull request merged**